js/stringProcessing/addWordboundary.js

```diff
@@ -0,0 +1,22 @@
+/** @module stringProcessing/addWordboundary */
+
+/**
+ * Returns a string that can be used in a regex to match a matchString with word boundaries.
```

Comment:
We should document here why this is a special word boundary better than the build in word boundaries of javascript. This makes sure that if someone poses the question "Why aren't we using build in word boundaries" that the answer is already documented.